docs/markdown/Dependencies.md

```diff
@@ -136,23 +136,49 @@ Since they can be used interchangeably, the rest of the build
 definitions do not need to care which one it is. Meson will take care
 of all the work behind the scenes to make this work.
 
-# Dependency method
+# Dependency detection method
 
 You can use the keyword `method` to let Meson know what method to use
 when searching for the dependency. The default value is `auto`.
-Additional dependencies methods are `pkg-config`, `config-tool`, `cmake`,
+Additional methods are `pkg-config`, `config-tool`, `cmake`,
 `builtin`, `system`, `sysconfig`, `qmake`, `extraframework` and `dub`.
 
 ```meson
 cups_dep = dependency('cups', method : 'pkg-config')
 ```
 
-The dependency method order for `auto` is:
+For dependencies without [specific detection
+logic](#dependencies-with-custom-lookup-functionality), the dependency method
+order for `auto` is:
 
   1. `pkg-config`
   2. `cmake`
   3. `extraframework` (OSX only)
-  4. `system`
+
+## System
+
+Some dependencies provide no valid methods for discovery, or do so only in
+some cases. Some examples of this are Zlib, which provides both pkg-config
+and cmake, except when it is part of the base OS image (such as in FreeBSD
+and macOS); OpenGL which has pkg-config on Unices from glvnd or mesa, but has
+no pkg-config on macOS and Windows.
+
+In these cases meson provides convenience wrappers in the form of `system`
+dependencies. Internally these dependencies do exactly what a user would do
+in the build system DSL or with a script, likely calling
+`compiler.find_library()`, setting `link_with` and `include_directories`. By
+putting these in meson upstream the barrier of using them is lowered, as
+projects using meson don't have to re-implement the logic.
```

Comment:
"Meson" is a proper name so it should be written properly capitalizes, not as "meson". There are multiple lower case spellings in this section.

Reply:
Done